winit/src/platform_impl/linux/x11/ime/context.rs

### Description

[{
	"resource": "/root/.cargo/registry/src/mirrors.ustc.edu.cn-4affec411d11e50f/winit-0.30.2/src/platform_impl/linux/x11/ime/context.rs",
	"owner": "rustc",
	"code": {
		"value": "Click for full compiler diagnostic",
		"target": {
			"$mid": 1,
			"path": "/diagnostic message [0]",
			"scheme": "rust-analyzer-diagnostics-view",
			"query": "0",
			"fragment": "file:///root/.cargo/registry/src/mirrors.ustc.edu.cn-4affec411d11e50f/winit-0.30.2/src/platform_impl/linux/x11/ime/context.rs"
		}
	},
	"severity": 8,
	"message": "mismatched types\nexpected fn pointer `unsafe extern \"C\" fn(_, *mut u8, *mut u8)`   found fn pointer `unsafe extern \"C\" fn(_, *mut i8, *mut i8)`",
	"source": "rustc",
	"startLineNumber": 161,
	"startColumn": 13,
	"endLineNumber": 163,
	"endColumn": 14
}]

### OS and window mananger

Kali Linux

### Winit version

0.30.2

I thought the same thing :D  I checked because it says Kali Linux, and wanted to make sure it wasn't something we are doing.

For what it's worth, the error when compiling alacritty is:

```
error[E0308]: mismatched types
   --> /home/steev/.cargo/registry/src/index.crates.io-6f17d22bba15001f/winit-0.30.2/src/platform_impl/linux/x11/ime/context.rs:161:13
    |
161 | /             mem::transmute::<usize, unsafe extern "C" fn(ffi::XIM, *mut i8, *mut i8)>(
162 | |                 preedit_start_callback as usize,
163 | |             )
    | |_____________^ expected `u8`, found `i8`
    |
    = note: expected fn pointer `unsafe extern "C" fn(_, *mut u8, *mut u8)`
               found fn pointer `unsafe extern "C" fn(_, *mut i8, *mut i8)`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `winit` (lib) due to 1 previous error
```

--- a/src/changelog/unreleased.md
+++ b/src/changelog/unreleased.md
@@ -50,3 +50,7 @@ changelog entry.
 - Remove `EventLoop::run`.
 - Remove `EventLoopExtRunOnDemand::run_on_demand`.
 - Remove `EventLoopExtPumpEvents::pump_events`.
+
+### Fixed
+
+- On X11, build on arm platforms.

--- a/src/platform_impl/linux/x11/ime/context.rs
+++ b/src/platform_impl/linux/x11/ime/context.rs
@@ -158,7 +158,7 @@ struct PreeditCallbacks {
 impl PreeditCallbacks {
     pub fn new(client_data: ffi::XPointer) -> PreeditCallbacks {
         let start_callback = create_xim_callback(client_data, unsafe {
-            mem::transmute::<usize, unsafe extern "C" fn(ffi::XIM, *mut i8, *mut i8)>(
+            mem::transmute::<usize, unsafe extern "C" fn(ffi::XIM, ffi::XPointer, ffi::XPointer)>(
                 preedit_start_callback as usize,
             )
         });

